docs/src/references/mappers/c8y-mapper.md

```diff
@@ -568,6 +568,305 @@ c8y/alarm/alarms/create
 
 </div>
 
+### Data
+
+The data telemetry type can be used to store additional information about the device/child/service. Such information could included; operation system name/version, communication statistics, device status, or any other information that is not suited to measurements, events or alarms.
+
+#### Data - Main device
+
+A device's digital twin model can be updated by publishing to a specific topic.
+
+The type part of the topic is used to group the data so it is easier for components to subscribe to relevant parts.
+
+<div class="code-indent-left">
+
+**Thin Edge (input)**
+
+```text title="Topic (retain=true)"
+te/device/main///data/device_OS
+```
+
+```json5 title="Payload"
+{
+  "family": "Debian",
+  "version": "11"
+}
+```
+
+</div>
+
+<div class="code-indent-right">
+
+**Cumulocity IoT (output)**
+
+```text title="Topic"
+c8y/inventory/managedObjects/update/<main-device-id>
+```
+
+```json5 title="Payload"
+{
+  "device_OS": {
+    "family": "Debian",
+    "version": "11"
+  }
+}
+```
+
+</div>
+
+
+#### Data - root
+
+Data can be added on the root level by leaving the type value in topic blank but still using a trailing slash.
+
+Some property names are forbidden and will be ignored if they are included in the payload. This is to avoid setting the property values from two different sources.
+
+* `name`
+* `type`
+
+Note: The payload should be restricted to the following data types:
+
+* string
+* boolean
+* number
+
+For complex data types like arrays and objects, these data types should be published to the typed topics.
+
+<div class="code-indent-left">
+
+**Thin Edge (input)**
+
+```text title="Topic (retain=true)"
+te/device/main///data/
+```
+
+```json5 title="Payload"
+{
+  "subtype": "my-custom-type"
+}
+```
+
+</div>
+
+<div class="code-indent-right">
+
+**Cumulocity IoT (output)**
+
+```text title="Topic"
+c8y/inventory/managedObjects/update/<main-device-id>
+```
+
+```json5 title="Payload"
+{
+  "subtype": "my-custom-type"
+}
+```
+
+</div>
+
+#### Data - Child Device
+
+<div class="code-indent-left">
+
+**Thin Edge (input)**
+
+```text title="Topic (retain=true)"
+te/device/child01///data/device_OS
+```
+
+```json5 title="Payload"
+{
+  "family": "Debian",
+  "version": "11"
+}
+```
+
+</div>
+
+<div class="code-indent-right">
+
+**Cumulocity IoT (output)**
+
+```text title="Topic"
+c8y/inventory/managedObjects/update/<main-device-id>:device:child01
+```
+
+```json5 title="Payload"
+{
+  "device_OS": {
+    "family": "Debian",
+    "version": "11"
+  }
+}
+```
+
+</div>
+
+#### Data - Service on Main Device
+
+<div class="code-indent-left">
+
+**Thin Edge (input)**
+
+```text title="Topic (retain=true)"
+te/device/main/service/tedge-agent/data/runtime_stats
```

Comment:
I understand that this is just an example, but realistically for things like `runtime_stats`, isn't it better to model them as measurements of that service, as it fairly dynamic in nature?

Reply:
It depends on the use-case. In Cumulocity IoT publishing some runtime statistics (at long intervals) is more useful as you can then query the devices based on these statistics. For example if you had a runtime statistics which sent the connection stability (as a percentage), then you can create a smart group in Cumulocity for devices which have a higher connection stability to then send the software updates to only these devices (or creates groups based on quota)...Currently Cumulocity IoT devices does not support filtering devices based on an event/measurement or alarm (except for the newish "last known value" feature which just converts the measurements to an inventory managed object fragment).

TLDR; Some use cases are:

* Allow filtering devices based on the published statistics
* Data is not subject to retention rules (otherwise you would loose the data after the retention period)